server/lib/saveImage.js

```diff
@@ -11,17 +15,15 @@ const decodeBase64Image = (data) => {
     return new Error(`Invalid input string: ${matches}`);
   }
 
-  return {
-    type: matches[1],
-    data: Buffer.from(matches[2], 'base64')
-  };
+  return Buffer.from(matches[2], 'base64');
 };
 
-const saveImage = (payload) => {
-  const imageBuffer = decodeBase64Image(payload);
-  const filename = `cropped.jpg`;
+const saveImage = (payload, filename) => {
```

Comment:
Changes to handle either saving the image coming from the webcam, or the gif coming from p5.